### DISCUSSION TITLE
Deal with a method of the deployed escrow contract within my smart contract

### DISCUSSION THREAD

**Discussion by Bassel-A-Elazab** [celo-org/social-connect#216] [Help (Q&A)] [answered]
I build my smart contract and I want to deal with the deployed Escrow contract, especially the transfer method.
I mean I want to transfer from my smart contract to pay the amount from it and not from my connected wallet. As we did in ERC20 tokens.
I found two ways but it failed.
1 - I build my smart contract and I accessed the Escrow within my contact by its address. I can deal with all the cal methods but I can't use the transfer.
2 - I found that I can use a method called abi.encodewithsignature. I used it but it send the transfer to my contract address and not to the Escrow.
In the final all I need instead of paying from my connected wallet I want to pay from my smart contract. But, I can't figure out what should I do especially there are many variables in Escrow should I deal with it and I want to use the ready deployed one.
If possible, I appreciated your help.
If it isn't possible, then I must deploy our Escrow contract. At this point, I can't use the Valora app in the withdrawal process and I build it with myself? This is correct!
Thanks!

**Comment by critesjosh** [comment, marked as answer]
I believe that using the escrow contract requires registering a phone number with an account on ODIS. You might want to create your own escrow contract depending on what you are trying to do. What is your use case?

**Reply by Bassel-A-Elazab** [reply]
I used it successfully! 
I send tokens to the Escrow contract and registered my number and I withdraw the token as I expected.
The only thing I need is instead of paying the amount that I want to transfer from the connected wallet I need to pay it from my smart contract. In this case, I should build my Escrow contract? If I should build it, AM I follow the Escrow contract implementation in the celo-monorepo.
Thanks!

**Reply by critesjosh** [reply]
You should be able to use the existing escrow contract, you will just need to make sure that your smart contract calls [this transfer](https://github.com/celo-org/celo-monorepo/blob/83b5c21c20fb22d17c46f73bc67b3de88f69e171/packages/protocol/contracts/identity/Escrow.sol#L110) function on the escrow.sol contract and that your contract has a balance of the tokens being escrowed.

**Reply by Bassel-A-Elazab** [reply]
Can you recommend a way that I can call the transfer method of the Escrow into my smart contract?
Because I knew the only way of abi.encodewithsignature and when I deal with it. it doesn't transfer value to the Escrow because of this line of the deployed Escrow.

require(ERC20(token).transferFrom(msg.sender, address(this), value), "Transfer unsuccessful.");

it returns back to my contract!
I appreciated your help!
Thanks!

**Reply by critesjosh** [reply]
if you click the link to the Escrow contract that I shared in the previous comment you will see that the interface for the `transfer` function in that contract is not the standard ERC20 interface. You will have to import the Escrow contract into yours to encode the function call appropriately. Do something like the following in your contract
```solidity
Escrow(escrowAddress).transfer(bytes32 identifier,
    address token,
    uint256 value,
    uint256 expirySeconds,
    address paymentId,
    uint256 minAttestations)
 ```

**Reply by Bassel-A-Elazab** [reply]
Thanks so much!
Much appreciated!
That thing I want and I followed it previously. But, I was using remix IDE and I can't use the transfer method. 
Then, I used truffle with your feedback and It worked fine as I expected and it succeeded.
and I received all funds after I download the Valora app. :slightly_smiling_face:
